servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/single/SupplierSingleTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © 2019 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.concurrent.api.single;
+
+import io.servicetalk.concurrent.SingleSource;
+import io.servicetalk.concurrent.api.Single;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.function.Supplier;
+
+import static io.servicetalk.concurrent.api.SourceAdapters.toSource;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+public class SupplierSingleTest {
```

Comment:
For every new source we create, we also write reactive streams TCK test. Please add those for the new implementations => `CallableSingle` and `RunnableCompletable`. As a reference see:

https://github.com/apple/servicetalk/blob/master/servicetalk-concurrent-reactivestreams/src/test/java/io/servicetalk/concurrent/reactivestreams/tck/SingleDeferTckTest.java

Reply:
will do